yjit/src/backend/x86_64/mod.rs

```diff
@@ -465,15 +465,16 @@ impl Assembler
                 // This assumes only load instructions can contain references to GC'd Value operands
                 Insn::Load { opnd, out } |
                 Insn::LoadInto { dest: out, opnd } => {
-                    mov(cb, out.into(), opnd.into());
-
-                    // If the value being loaded is a heap object
-                    if let Opnd::Value(val) = opnd {
-                        if !val.special_const_p() {
+                    match opnd {
+                        // If the value being loaded is a heap object
+                        Opnd::Value(val) if !val.special_const_p() => {
```

Comment:
Maybe we should add a `heap_object_p()` ? I feel like it would be more readable than `!val.special_const_p()`

Reply:
:+1: https://github.com/ruby/ruby/pull/6733/commits/8ee46230a8d69034c52678fd45486437fb170126